### PR TITLE
Wrap pkg-config

### LINKED ISSUE #86077: pkg-config needs prefixes and wrappper like cc and binutils
Pkg-config needs a wrapper, so we we have `${target-prefix}` pkg-config and `BUILD_PKG_CONFIG_PATH` environment variables.

CC @matthewbauer

### PR DESCRIPTION
**Pull request by Ericson2314**
###### Motivation for this change

Fix #86077

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/test/cross/default.nix [hunk 1]
```diff
@@ -91,6 +91,19 @@ let
...
+    {
+      depsBuildBuild = [ crossPkgs.pkgsBuildBuild.pkg-config ];
+      nativeBuildInputs = [ crossPkgs.pkgsBuildHost.pkg-config crossPkgs.pkgsBuildHost.zlib ];
+      depsBuildTarget = [ crossPkgs.pkgsBuildTarget.pkg-config ];
+      buildInputs = [ crossPkgs.zlib ];
+    } ''
+       mkdir $out
+       ${crossPkgs.pkgsBuildBuild.pkg-config.targetPrefix}pkg-config --cflags zlib > "$out/for-build"
+       ${crossPkgs.pkgsBuildHost.pkg-config.targetPrefix}pkg-config --cflags zlib > "$out/for-host"
+       !diff "$out/for-build" "$out/for-host"
```
Comment (matthewbauer) [inline pkgs/test/cross/default.nix:105]: this looks like a mistake

Comment (Ericson2314) [reply pkgs/test/cross/default.nix:105]: I wanted to see that they are different, though I should probably improve the test to raise things to a higher bar.

Comment (matthewbauer) [reply pkgs/test/cross/default.nix:105]: what does ! do?

Comment (Ericson2314) [reply pkgs/test/cross/default.nix:105]: negate the exit code, at least that is what I intend.

**Comment by Ericson2314** [comment]
The LLVM test's dep's failure is unrelated, and the gcc one past locally.

File: pkgs/test/cross/default.nix (hunk 1 above)
Comment (Ericson2314) [reply pkgs/test/cross/default.nix]: Ah, I needed the space `! diff`

**Comment by FRidh** [comment]
Discussion regarding failure on staging https://github.com/NixOS/nixpkgs/issues/89170

**Comment by FRidh** [comment]
Cross-compilation failure is now blocking staging-next https://github.com/NixOS/nixpkgs/pull/89388#issuecomment-639155770.

**Pull request merged**